Project/Source/Components/Trail.cpp

```diff
@@ -0,0 +1,298 @@
+			}
+		}
+	}
+}
+
+void Trail::UpdateLife(Quad* currentQuad) {
+	if (App->time->IsGameRunning()) {
+		currentQuad->life -= App->time->GetDeltaTime();
+	} else {
+		currentQuad->life -= App->time->GetRealTimeDeltaTime();
+	}
+}
+
+void Trail::OnEditorUpdate() {
+	if (ImGui::Button("Play")) Play();
+	if (ImGui::Button("Stop")) Stop();
+	if (ImGui::Checkbox("Render", &isRendering)) {
+		isStarted = false;
+	}
+	ImGui::DragFloat("Witdh", &width, App->editor->dragSpeed2f, 0, inf);
+	if (ImGui::DragInt("Trail Quads", &trailQuads, 1.0f, 1, 50, "%d", ImGuiSliderFlags_AlwaysClamp)) {
+		if (nTextures > trailQuads) nTextures = trailQuads;
+		DeleteQuads();
+	}
+	if (ImGui::DragInt("Texture Repeats", &nTextures, 1.0f, 1, trailQuads, "%d", ImGuiSliderFlags_AlwaysClamp)) {
+		DeleteQuads();
+		EditTextureCoords();
+	}
+
+	ImGui::DragFloat("Quad Life", &quadLife, App->editor->dragSpeed2f, 1, inf);
+
+	ImGui::Checkbox("Color Over Trail", &colorOverTrail);
+	if (colorOverTrail) {
+		ImGui::GradientEditor(gradient, draggingGradient, selectedGradient);
+	}
+
+	UID oldID = textureID;
+	ImGui::ResourceSlot<ResourceTexture>("texture", &textureID);
+	ResourceTexture* textureResource = App->resources->GetResource<ResourceTexture>(textureID);
+	if (textureResource != nullptr) {
+		int width;
+		int height;
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_WIDTH, &width);
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_HEIGHT, &height);
+
+		ImGui::NewLine();
+		ImGui::Separator();
+		ImGui::TextColored(App->editor->titleColor, "Texture Preview");
+		ImGui::TextWrapped("Size:");
+		ImGui::SameLine();
+		ImGui::TextWrapped("%d x %d", width, height);
+		ImGui::Image((void*) textureResource->glTexture, ImVec2(200, 200));
+		ImGui::Separator();
+	}
+}
+
+void Trail::Draw() {
+	ProgramTrail* trailProgram = App->programs->trail;
+	if (!trailProgram) return;
+	unsigned glTexture = 0;
+
+	ResourceTexture* texture = App->resources->GetResource<ResourceTexture>(textureID);
+	glTexture = texture ? texture->glTexture : 0;
+	int hasDiffuseMap = texture ? 1 : 0;
+
+	glDepthMask(GL_FALSE);
+	glEnable(GL_BLEND);
+	glBlendEquation(GL_FUNC_ADD);
+	glBlendFunc(GL_ONE, GL_ONE);
+	glDisable(GL_CULL_FACE);
+
+	for (int i = 0; i < quadsCreated; i++) {
+		Quad* currentQuad = &quads[i];
+
+		glBindBuffer(GL_ARRAY_BUFFER, quadVBO); // set vbo active
+		glBufferData(GL_ARRAY_BUFFER, sizeof(currentQuad->quadInfo), currentQuad->quadInfo, GL_STATIC_DRAW);
+
+		glEnableVertexAttribArray(0);
+		glEnableVertexAttribArray(1);
+		glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, sizeof(float) * 5, (void*) 0);
+		glVertexAttribPointer(1, 2, GL_FLOAT, GL_FALSE, sizeof(float) * 5, (void*) (sizeof(float) * 3));
+		glUseProgram(trailProgram->program);
+
+		Frustum* frustum = App->camera->GetActiveCamera()->GetFrustum();
+		float4x4* proj = &App->camera->GetProjectionMatrix();
+		float4x4* view = &App->camera->GetViewMatrix();
+
+		glActiveTexture(GL_TEXTURE0);
```

Comment:
This is being called twice. It's not going to be a problem, but I think it would be better to remove it.